src/lib/stores.ts

```diff
@@ -15,15 +19,68 @@ export const inSnsMenu = writable<boolean>(false);
 /* === Components === */
 export const language = writable<'en' | 'es' | 'ja' | 'ru'>('en');
 export const selectedAsset = writable<Asset>(new Asset(AssetType.ICP));
-export const selectedSns = writable<string>('BOOM DAO');
-export const snsPrincipal = writable<string>('xomae-vyaaa-aaaaq-aabhq-cai');
-export const encodedSnsIcrcAccount = writable<string>('-/-');
 export const user = writable<User | undefined>(undefined);
 export const canisters = writable<Canisters | undefined>(undefined);
 export const waterNeuronInfo = writable<WaterNeuronInfo | undefined>(undefined);
 
+/* === SNS === */
+function createSns() {
+	const { subscribe, set, update } = writable<{
+		name: string;
+		principal: string;
+		encodedAccount: string;
+		icpBalance: BigNumber | undefined;
+		nicpBalance: BigNumber | undefined;
+	}>({
+		name: '',
+		principal: '',
+		encodedAccount: '-/-',
+		icpBalance: undefined,
+		nicpBalance: undefined
+	});
+
+	return {
+		subscribe,
+		setPrincipal: (principal: string) => update((sns) => ({ ...sns, principal })),
+		setName: (name: string) => update((sns) => ({ ...sns, name })),
+		setEncodedAccount: (encodedAccount: string) => update((sns) => ({ ...sns, encodedAccount })),
+		setIcpBalance: (icpBalance: BigNumber) => update((sns) => ({ ...sns, icpBalance })),
+		setNicpBalance: (nicpBalance: BigNumber) => update((sns) => ({ ...sns, nicpBalance })),
+		reset: () =>
+			set({
+				name: '',
+				principal: '',
+				encodedAccount: '-/-',
+				icpBalance: undefined,
+				nicpBalance: undefined
+			})
+	};
+}
+export const sns = createSns();
+
+export async function handleSnsChange(name?: string, principal?: string) {
```

Comment:
We you don't fetch the balance in this function? Look at https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise to make calls in parallel.

Reply:
Because I call this function only when an Sns is selected but I fetch the balances every 5 seconds.

Reply:
It's true that the probability that the SNS treasury funds vary every 5 seconds is low

Reply:
I think we can fetch it once.